src/Simple.OData.Client.Core/Fluent/FluentCommand.cs

```diff
@@ -579,30 +606,25 @@ private string Format()
                     return null;
 
                 var keyNames = _details.Session.Metadata.GetDeclaredKeyPropertyNames(this.EntityCollection.Name).ToList();
-                var namedKeyValues = new Dictionary<string, object>();
-                for (var index = 0; index < keyNames.Count; index++)
+
+                return (_details.NamedKeyValues?.Select(x => new
```

Comment:
Out of curiosity, why was this changed from the original implementation?

Reply:
The original implementation was:
- When named values given, filter out invalid names using the primary key definition
- When values given, zip together with the primary key names

I tried to perform minimal changes so I tried to squeeze alternate keys in there too. The intent was to discover matching model fields so that I only keep the original behaviour for primary and not alternate keys. Looking at it again I forgot that filter so I commited a fix and relevant tests for that, replicating the original behaviour (filter out invalid names when primary key). If we perform the validation discussed on the previous review item then this can go.